plugins/modules/notification/mattermost.py

```diff
@@ -38,7 +38,14 @@
     type: str
     description:
       - Text to send. Note that the module does not handle escaping characters.
-    required: true
+      - Required when I(attachments) is not set.
+  attachments:
+    type: list
+    elements: dict
+    description:
+      - Define a list of attachments.
+      - For more information, see U(https://developers.mattermost.com/integrate/admin-guide/admin-message-attachments/).
+      - Required when I(text) is not set.
```

Comment:
Right now it is **not** required when `text` is not set. You need to add a `required_one_of` parameter to the `AnsibleModule()` constructor call (grep for it in plugins/modules/ to find usage examples). Also you need to remove the `required: true` from the `text` option in the documentation as well (that's why the sanity tests currently fail); you should also update its description to mention that at least one of `text` and `attachments` needs to be specified.

Reply:
Thanks, for the suggestion i changed this.